internal/types/action/action.go

```diff
@@ -8,110 +8,114 @@ import (
-	DeleteSelf                Type = 34
-	NoOp                      Type = 35
-	AddCredentialLibraries    Type = 36 // DEPRECATED
-	SetCredentialLibraries    Type = 37 // DEPRECATED
-	RemoveCredentialLibraries Type = 38 // DEPRECATED
-	AddCredentialSources      Type = 39
-	SetCredentialSources      Type = 40
-	RemoveCredentialSources   Type = 41
-	AddHostSources            Type = 42
-	SetHostSources            Type = 43
-	RemoveHostSources         Type = 44
-	CreateWorkerLed           Type = 45
-	AddWorkerTags             Type = 46
-	SetWorkerTags             Type = 47
-	RemoveWorkerTags          Type = 48
+	Unknown                          Type = 0
+	List                             Type = 1
+	Create                           Type = 2
+	Update                           Type = 3
+	Read                             Type = 4
+	Delete                           Type = 5
+	Authenticate                     Type = 6
+	All                              Type = 7
+	AuthorizeSession                 Type = 8
+	AddGrants                        Type = 9
+	RemoveGrants                     Type = 10
+	SetGrants                        Type = 11
+	AddPrincipals                    Type = 12
+	SetPrincipals                    Type = 13
+	RemovePrincipals                 Type = 14
+	Deauthenticate                   Type = 15
+	AddMembers                       Type = 16
+	SetMembers                       Type = 17
+	RemoveMembers                    Type = 18
+	SetPassword                      Type = 19
+	ChangePassword                   Type = 20
+	AddHosts                         Type = 21
+	SetHosts                         Type = 22
+	RemoveHosts                      Type = 23
+	AddHostSets                      Type = 24 // DEPRECATED
+	SetHostSets                      Type = 25 // DEPRECATED
+	RemoveHostSets                   Type = 26 // DEPRECATED
+	Cancel                           Type = 27
+	AddAccounts                      Type = 28
+	SetAccounts                      Type = 29
+	RemoveAccounts                   Type = 30
+	ReadSelf                         Type = 31
+	CancelSelf                       Type = 32
+	ChangeState                      Type = 33
+	DeleteSelf                       Type = 34
+	NoOp                             Type = 35
+	AddCredentialLibraries           Type = 36 // DEPRECATED
+	SetCredentialLibraries           Type = 37 // DEPRECATED
+	RemoveCredentialLibraries        Type = 38 // DEPRECATED
+	AddCredentialSources             Type = 39
+	SetCredentialSources             Type = 40
+	RemoveCredentialSources          Type = 41
+	AddHostSources                   Type = 42
+	SetHostSources                   Type = 43
+	RemoveHostSources                Type = 44
+	CreateWorkerLed                  Type = 45
+	AddWorkerTags                    Type = 46
+	SetWorkerTags                    Type = 47
+	RemoveWorkerTags                 Type = 48
+	ReinitializeCertificateAuthority Type = 49
+	ReadCertificateAuthority         Type = 50
```

Comment:
You didn't follow the comments below! :-)

Reply:
I think it's asking for an update to the wrong test:
https://github.com/hashicorp/boundary/blob/main/internal/perms/acl_test.go#L391

vs what I think it should be, TestActions
https://github.com/hashicorp/boundary/blob/main/internal/types/action/action_test.go#L9

I had looked at Test_AnonRestrictions and assumed I was covered since I thought it should fail for the anon user. 
(Please let me know if I'm misunderstanding anything)

Reply:
It's missing from TestActions too! But no, it's correct about Test_AnonRestrictions. In the for loop in the test, it iterates based on the numeric value of the action. Right now it maxes out at RemoveHostSources; it needs to be upped to max out at ReadCertificateAuthority.

(There's an additional bug in that test in that the resource wasn't updated to max out at Credential instead of CredentialLibrary).

Reply:
Ohhh I see that now. I'll fix those.